mgmt: mcumgr: transport: smp: Fix smp_init runlevel.

bas-archembedded opened the pull request:
Move priority to POST_KERNEL to make sure smp_init() is called before smp_init_uart as it initializes the transport list in case CONFIG_SMP_CLIENT is enabled.

The smp_init() routine was called after smp_init_uart(), causing the list to be emptied again after registration of the uart client transport.

nordicjm reviewed (changes requested):
This is not correct, `APPLICATION` is correct. If there is an order that is needed than you could add Kconfigs to specify the run levels

bas-archembedded committed:
mgmt: mcumgr: transport: smp: Fix smp_transport_clients list init.

Make sure smp_transport_clients list is only initialized once and
before any transports will try to register and add entries to this
list.

The smp_init() routine was called after smp_init_uart(), causing
the list to be emptied again after registration of the uart client
transport.

Signed-off-by: Bas van Loon <bas@arch-embedded.com>

nordicjm reviewed (commented):
As an alternative without having to mess with init levels which seems messy, have you tried having the slist init'd statically instead using https://docs.zephyrproject.org/latest/doxygen/html/group__single-linked-list__apis.html#ga7f4710125f45643b7acdaa58dbfff225 ?
